koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -945,3 +954,271 @@ def handler(self, src, target, opts=None):
+            except Exception as error:
+                self.logger.info("Error while saving incremental logs: %s", error)
+                os._exit(1)
+            os._exit(0)
+
+        response = self.osbs().wait_for_build_to_finish(build_id)
+
+        self.logger.debug("OSBS build finished with status: %s. Build "
+                          "response: %s.", response.status,
+                          response.json)
+
+        self.logger.info("Response status: %r", response.is_succeeded())
+
+        if response.is_cancelled():
+            self.session.cancelTask(self.id)
+            raise ContainerCancelled(
+                'Image build was cancelled by OSBS, maybe by automated rebuild.'
+            )
+
+        elif response.is_failed():
+            error_message = self._get_error_message(response)
+            if error_message:
+                raise ContainerError('Image build failed. %s. OSBS build id: %s' %
+                                     (error_message, build_id))
+            else:
+                raise ContainerError('Image build failed. OSBS build id: %s' %
+                                     build_id)
+
+        repositories = []
+        if response.is_succeeded():
+            repositories = self._get_repositories(response)
+
+        self.logger.info("Image available in the following repositories: %r",
+                         repositories)
+
+        koji_build_id = None
+        if response.is_succeeded():
+            koji_build_id = self._get_koji_build_id(response)
+
+        self.logger.info("Koji content generator build ID: %s", koji_build_id)
+
+        containerdata = {
+            'task_id': self.id,
+            'osbs_build_id': build_id,
+            'files': [],
+            'repositories': repositories,
+            'koji_build_id': koji_build_id,
+        }
+
+        return containerdata
+
+    def get_source_build_info(self, build_id, build_nvr):
+        build_identifier = build_nvr or build_id
+
+        koji_build = self.session.getBuild(build_identifier)
+        if not koji_build:
+            raise koji.BuildError("specified source build '%s' doesn't exist" % build_identifier)
+
+        if build_id and (build_id != koji_build['build_id']):
+            err_msg = (
+                'koji_build_id {} does not match koji_build_nvr {} with id {}. '
+                'When specifying both an id and an nvr, they should point to the same image build'
+                .format(build_id, build_nvr, koji_build['build_id'])
+                )
+            raise koji.BuildError(err_msg)
+
+        if not build_id:
+            build_id = koji_build['build_id']
+        if not build_nvr:
+            build_nvr = koji_build['nvr']
+        component = "{}-source" % koji_build['package_name']
+
+        return component, build_id, build_nvr
+
+    def handler(self, target, opts=None):
+        jsonschema.validate([target, opts], self.PARAMS_SCHEMA)
+
+        if opts is None:
+            opts = {}
```

Comment:
It can't be `None` if it passed validation.

Reply:
right